aspnetcore/razor-pages/razor-pages-conventions.md

```diff
@@ -63,6 +63,24 @@ public void ConfigureServices(IServiceCollection services)
 }
 ```
 
+## Route order
+
+Routes specify an <xref:Microsoft.AspNetCore.Mvc.ApplicationModels.AttributeRouteModel.Order*> for processing (route matching).
+
+| Order            | Behavior |
+| :--------------: | -------- |
+| -1               | The route is processed before other routes are processed. |
+| 0                | Order isn't specified (default value). Not assigning `Order` (`Order = null`) defaults the route `Order` to 0 (zero) for processing. |
+| 1, 2, &hellip; n | Specifies the route processing order. |
+
+Route processing is established by convention:
+
+* Routes are processed in sequential order (-1, 0, 1, 2, &hellip; n).
+* When routes have the same `Order`, the most specific route is matched first followed by less specific routes.
+* When routes with the same `Order` and the same number of parameters match a request URL, routes are processed in the order that they're added to the <xref:Microsoft.AspNetCore.Mvc.ApplicationModels.PageConventionCollection>.
+
+If possible, avoid depending on an established route processing order. Generally, routing selects the correct route with URL matching. If you must set route `Order` properties to route requests correctly, the app's routing scheme is probably confusing to clients and fragile to maintain. Seek to simplify the app's routing scheme. The sample app requires an explicit route processing order to demonstrate several routing scenarios using a single app. However, you should attempt to avoid the practice of setting route `Order` in production apps.
```

Comment:
This section all seems pretty good. If there's details about this in the regular routing section it might be helpful to link it. This is a shared implementation with regular routing.

Reply:
Yes, there's one spot in routing where this is discussed, and there is some duplication. Of course, `PageCollectionConvention` isn't discussed over there. I'll take another look at this in a few minutes and see about cross-linking both ways (just in case someone looking for RP content lands over there).